### PR TITLE
Feature/exercise links in readings

### PR DESCRIPTION
**Pull request by pandafulmanda**
screen shot 2015-07-17 at 7 39 30 pm

CORS is an issue.  Need to do this to Chrome to load in exercises:
```bash
open -a Google\ Chrome --args --allow-file-access-from-files --disable-web-security
```

- [x] Make processLink in ```book-content-mixin``` less atrocious.
  * pulling out if else stuff using filter or something similar to group types of links first and then processing
- [ ] ~~style updates as needed~~, separate PR
- [X] hide ```ost-reading-discard```, already hidden

### CODE REVIEW

**Comment by philschatz** [comment]
Does the CORS stuff need to be added to `tutor-server` to allow access to the appropriate exercises server?

**Comment by pandafulmanda** [comment]
I talked to @Dantemss and @jpslav about CORS, and they will opening it up on the openstax/exercises server

**Comment by philschatz** [comment]
has the BE added exercises to the CORS list? @jpslav @Dantemss ?

File: src/components/book-content-mixin.cjsx [hunk 1]
```diff
@@ -63,41 +63,46 @@ module.exports =
     mediaLink.href = pageUrl + mediaLink.hash
     mediaLink.target = '_blank' if @shouldOpenNewTab?()
 
-  processLink: (mediaLink) ->
-    return unless @isMediaLink(mediaLink)
+  isMediaOnPage: (link) ->
     root = @getDOMNode()
+    try
```
Comment (philschatz) [inline src/components/book-content-mixin.cjsx]: Why is this `try` needed?

File: src/components/reference-book/page.cjsx [hunk 2]
```diff
@@ -78,6 +83,30 @@ module.exports = React.createClass
       for image in images
         image.addEventListener('load', onImageLoad)
 
+  isExerciseLink: (link) ->
+    link.hash.search(EXERCISE_MATCHER) is 0
```
Comment (philschatz) [inline src/components/reference-book/page.cjsx]: Would just `EXERCISE_MATCHER.test(link.hash)` work?

File: src/components/reference-book/page.cjsx [hunk 3]
```diff
@@ -78,6 +83,30 @@ module.exports = React.createClass
...
+  renderOtherLinks: (otherLinks) ->
+    ReferenceBookExerciseStore.setMaxListeners(otherLinks.length)
+    _.each(otherLinks, @renderExercise)
+
+  renderExercise: (link) ->
+    if @isExerciseLink(link)
+      exerciseId = @getExerciseId(link)
+      if link.parentNode.parentNode?
+        @_exerciseNodes.push(link.parentNode.parentNode)
+        React.render(<ReferenceBookExerciseShell exerciseId={exerciseId}/>, link.parentNode.parentNode)
```
Comment (philschatz) [inline src/components/reference-book/page.cjsx]: Oh, nice; nested `React.render` ; )

Comment (philschatz) [inline src/components/reference-book/page.cjsx]: Does the unmount also need to occur when the page component updates?

File: src/components/reference-book/page.cjsx (hunk 2 above)
Comment (pandafulmanda) [inline src/components/reference-book/page.cjsx]: yeah i just wanted to be as specific as possible

**Pull request merged**